libfwupdplugin/fu-device-locker.c

```diff
@@ -202,8 +202,14 @@ fu_device_locker_new_full (gpointer device,
 	self->close_func = close_func;
 
 	/* open device */
-	if (!self->open_func (device, error))
+	if (!self->open_func (device, error)) {
+		g_autoptr(GError) error_local = NULL;
+		if (!self->close_func (device, &error_local)) {
+			g_debug ("ignoring close error on aborted open: %s",
+				 error_local->message);
+		}
```

Comment:
devil's advocate here - why shouldn't this be part of finalizing the device?

Reply:
I did think about that, but then we also encourage retrying an action if it fails. e.g. if we retry ->open 3 times, then fail in setup() (in the last step of `fu_device_open()`, possibly after doing something like opening a USB device) and finalize, we'd expect to get 3 ->close's rather than just one.